src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -817,7 +817,7 @@ private void onStorageExpire(Iterable<Digest> digests) {
   private void blockUntilShutdown() throws InterruptedException {
     // should really be waiting for either server or pipeline shutdown
     try {
-      if (pipeline.hasStages()) {
+      if (pipeline.hasStages() || hasExecutionCapability) {
```

Comment:
The execution pipeline can fail on startup by being unable to connect to redis.  When that happens, the pipeline has no stages defined.  I'd like the worker to shutdown during this issue instead of block indefinitely.  That's the behavior we had previously, but I broke it when trying to avoid storage workers automatically shutting down by having no pipeline.